OptimizelySDK/Odp/OdpManager.cs

```diff
@@ -0,0 +1,296 @@
+        }
+
+        /// <summary>
+        /// Sends signal to stop Event Manager and clean up ODP Manager use
+        /// </summary>
+        public void Dispose()
+        {
+            if (EventManager == null || !_enabled)
+            {
+                return;
+            }
+
+            EventManager.Stop();
+        }
+
+        /// <summary>
+        /// Builder pattern to create an instances of OdpManager
+        /// </summary>
+        public class Builder
+        {
+            private OdpConfig _odpConfig;
+            private IOdpEventManager _eventManager;
+            private IOdpSegmentManager _segmentManager;
+            private int _cacheSize;
+            private int _cacheTimeoutSeconds;
+            private ILogger _logger;
+            private IErrorHandler _errorHandler;
+            private ICache<List<string>> _cache;
+
+            public Builder WithSegmentManager(IOdpSegmentManager segmentManager)
+            {
+                _segmentManager = segmentManager;
+                return this;
+            }
+
+            public Builder WithEventManager(IOdpEventManager eventManager)
+            {
+                _eventManager = eventManager;
+                return this;
+            }
+
+            public Builder WithOdpConfig(OdpConfig odpConfig)
+            {
+                _odpConfig = odpConfig;
+                return this;
+            }
+
+            public Builder WithCacheSize(int cacheSize)
+            {
+                _cacheSize = cacheSize;
+                return this;
+            }
+
+            public Builder WithCacheTimeout(int seconds)
+            {
+                _cacheTimeoutSeconds = seconds;
+                return this;
+            }
+
+            public Builder WithLogger(ILogger logger = null)
+            {
+                _logger = logger;
+                return this;
+            }
+
+            public Builder WithErrorHandler(IErrorHandler errorHandler = null)
+            {
+                _errorHandler = errorHandler;
+                return this;
+            }
+
+            public Builder WithCacheImplementation(ICache<List<string>> cache)
+            {
+                _cache = cache;
+                return this;
+            }
+
+            /// <summary>
+            /// Build OdpManager instance using collected parameters
+            /// </summary>
+            /// <param name="asEnabled">Should mark as enabled upon initialization</param>
+            /// <returns>OdpManager instance</returns>
+            public OdpManager Build(bool asEnabled = true)
+            {
+                var manager = new OdpManager
+                {
+                    _odpConfig = _odpConfig,
+                    _logger = _logger ?? new NoOpLogger(),
+                    _enabled = asEnabled,
+                };
+
+                _errorHandler = _errorHandler ?? new NoOpErrorHandler();
+
+                if (_eventManager == null)
```

Comment:
if _enabled == false, then we do not need instantiate EventManager and SegmentManager.